### PR TITLE
OCPBUGS-179: fix conditional edge for type always

### PR DESCRIPTION
**Pull request by PratikMahajan**
conditional edge has an empty promql when the type of
conditional block is `Always`
This commit skips the serlialization if promql if the
promql condition is empty, which is the case for
type `Always`

Co-Authored-By: W. Trevor King <wking@tremily.us>

### CODE REVIEW

File: cincinnati/src/lib.rs [hunk 1]
```diff
@@ -1169,13 +1193,22 @@ mod tests {
 
     #[test]
     fn serialize_graph_with_conditional_edges() {
-        let graph = generate_graph(true);
+        let graph = generate_graph(true, false);
         assert_eq!(
             serde_json::to_string(&graph).unwrap(),
             r#"{"nodes":[{"version":"1.0.0","payload":"image/1.0.0","metadata":{}},{"version":"2.0.0","payload":"image/2.0.0","metadata":{}},{"version":"3.0.0","payload":"image/3.0.0","metadata":{}}],"edges":[[0,1],[1,2],[0,2]],"conditionalEdges":[]}"#
```
Comment (wking) [inline cincinnati/src/lib.rs]: `"conditionalEdges":[]` is not a very interesting set of conditional edges.  Can we use a PromQL example in this `generate_graph` case?

File: cincinnati/src/lib.rs [hunk 2]
```diff
@@ -818,12 +818,36 @@ pub mod testing {
...
 
-    pub fn generate_graph(include_conditional_edge: bool) -> Graph {
+    pub fn generate_graph(include_conditional_edge: bool, include_always_condition: bool) -> Graph {
         let mut graph = Graph::default();
 
+        if include_conditional_edge && include_always_condition {
+            graph.conditional_edges = Some(vec![ConditionalEdge {
+                edge_regex: ConditionalUpdateEdge {
+                    from: "1.0.0".to_string(),
+                    to: "2\\.0\\.*".to_string(),
```
Comment (wking) [inline cincinnati/src/lib.rs]: This expression should probably be `2[.]0[.].*`.  `[.]` is easier for me to think about than stacking escapes with `\\.`.  And you need an un-escaped `.` for the `.*` portion.

Comment (wking) [reply cincinnati/src/lib.rs]: err, also, when we declare conditional risks in graph-data, we have a literal `to` and a regular expression `from`.  In this code at the moment, you have a literal `from` and a regular expression `to`, which seems backwards.

File: cincinnati/src/lib.rs [hunk 3]
```diff
@@ -818,12 +818,36 @@ pub mod testing {
...
+                    from: "1.0.0".to_string(),
+                    to: "2.0.0".to_string(),
+                }],
+                risks: vec![ConditionalUpdateRisk {
+                    url: "https://bug.example.com/show_bug.cgi?id=example".to_string(),
+                    name: "AllBrokenUpdates".to_string(),
+                    message: "All Updates are broken".to_string(),
+                    matching_rules: vec![ClusterCondition {
+                        condition_type: "Always".to_string(),
+                        promql: PromQLClusterCondition {
```
Comment (wking) [inline cincinnati/src/lib.rs:840]: if we leave this unpopulated, what happens?

Comment (PratikMahajan) [reply cincinnati/src/lib.rs:840]: using `Default:default()` for `Always` condition

File: cincinnati/src/lib.rs (hunk 1 above)
Comment (PratikMahajan) [reply cincinnati/src/lib.rs]: addressed

File: cincinnati/src/lib.rs (hunk 2 above)
Comment (PratikMahajan) [reply cincinnati/src/lib.rs]: fixed

File: cincinnati/src/lib.rs [hunk 4]
```diff
@@ -821,28 +821,41 @@ pub mod testing {
...
         let mut graph = Graph::default();
 
-        if include_conditional_edge && include_always_condition {
-            graph.conditional_edges = Some(vec![ConditionalEdge {
+        if include_conditional_edge {
+            let mut ce = ConditionalEdge {
                 edge_regex: ConditionalUpdateEdge {
-                    from: "1.0.0".to_string(),
-                    to: "2\\.0\\.*".to_string(),
+                    from: "1[.]0[.]*".to_string(),
```
Comment (wking) [inline cincinnati/src/lib.rs]: still needs a `.*`, like `1[.]0[.].*`

**Review by wking — APPROVED** [review]
/lgtm

**Review by wking — APPROVED** [review]
/lgtm

**Comment by PratikMahajan** [comment]
/override ci/prow/e2e 

Will be fixed in https://github.com/openshift/cincinnati/pull/719

**Pull request merged**